app/validators/postal_code_validator.rb

```diff
@@ -12,11 +12,9 @@ def validate(record)
       return
     end
 
-    postal_code = record.postal_code.delete(' ').upcase
-
-    if country.eql?(ISO3166::Country[:us]) && !postal_code.match(ZIP_CODE_REGEX)
+    if country.eql?(ISO3166::Country[:us]) && !record.postal_code.match(ZIP_CODE_REGEX)
       record.errors[:postal_code] << "should be 12345 or 12345-1234"
-    elsif country.eql?(ISO3166::Country[:ca]) && !postal_code.match(POSTAL_CODE_REGEX)
+    elsif country.eql?(ISO3166::Country[:ca]) && !record.postal_code.match(POSTAL_CODE_REGEX)
```

Comment:
What do you think about extracting these two conditionals out to more intention revealing private predicate methods?  Like `valid_us_zip_code?` and `valid_us_postal_code?` ? It's self documenting and reduces the cognitive friction a little bit more.

Reply:
careful now, `valid_CA_postal_code?`

Reply:
Oopsies

Reply:
Do you mean just the latter half of the conditional, or the whole conditional?

The conditional is doing two distinct things - checking what country the record is describing, and then checking the validity of the postal code in that country. I personally don't think `valid_us_zip_code` makes that more clear, and the extra layer of abstraction obscures the meat of the code from what is already what feels to me like a manageable context. Naming a function `valid_ca_postal_code` feels redundant when it's in a class called `PostalCodeValidator`.

Maybe a rename of the constants would provide a similar benefit? `VALID_CANADIAN_POSTAL_CODE` and `VALID_US_ZIP_CODE` or something?

Reply:
The notion that POSTAL_CODE_REGEX = Canada and ZIP_CODE_REGEX = USA isn't obvious at least for me as a non frequent snail mailer to Canada.    So I'm a fan of renaming the Regex constants like @tobymurray mentioned.